src/test/scala/is/hail/utils/ArrayBuilderSuite.scala

```diff
@@ -8,7 +8,25 @@ class ArrayBuilderSuite extends SparkSuite {
     val ab = new ArrayBuilder[Int](0)
     ab += 3
     val a = ab.result()
-    assert(a.size == 1)
+    assert(a.length == 1)
     assert(a(0) == 3)
   }
+
+  @Test def addArray() {
+    val ab = new ArrayBuilder[Int](0)
+    ab ++= Array.fill[Int](5)(2)
+    val a = ab.result()
+    assert(a.length == 5)
+    assert(a.indices.forall(a(_) == 2))
+
+    val ab2 = new ArrayBuilder[Int](0)
+    ab2 ++= (Array.fill[Int](4)(3), 2)
+    val a2 = ab2.result()
+    assert(a2.length == 2)
+    assert(a2.indices.forall(a2(_) == 3))
+
+    ab2(0) = 5
```

Comment:
You're using apply here, not update. I think it will be simpler just to use `sameElements Array(5, 3)` rather than:
`ab2Update.zipWithIndex.forall { case (v, i) => if (i != 0) v == 3 else v == 5 }`

Reply:
It's calling update. Confirmed with the debugger. I think a(i) = j is syntactic sugar for a.update(i, j)